library/std/src/sys/unix/ext/net/ancillary.rs

```diff
@@ -240,25 +248,27 @@ impl<'a> AncillaryData<'a> {
     target_env = "uclibc",
 ))]
 #[unstable(feature = "unix_socket_ancillary_data", issue = "none")]
-impl<'a> AncillaryData<'a> {
-    fn from(cmsg: &'a libc::cmsghdr) -> Self {
+impl<'a> TryFrom<&'a libc::cmsghdr> for AncillaryData<'a> {
```

Comment:
API-wise I'm not sure it makes sense for this to be a TryFrom, rather than a method on AncillaryData. I expect we'll never have methods taking `T: TryInto<AncillaryData<'a>>` or anything like that ...

I am also unsure if this being a trait impl it might not be possible to make it truly unstable, i.e., it could possibly be observed to exist on stable. We probably don't want a public impl involving libc types in std.

Reply:
Okay, I will make this function to private method of `AncillaryData`.